File: SmartWash/src/main/java/com/smartwash/common/CouponStatus.java
package com.smartwash.common;

import lombok.Getter;

@Getter
public enum CouponStatus {
    EXPIRED(1, "已失效"),
    ACTIVE(0, "生效中"),
    RECEIVE(2, "已领取");

    private final Integer status;

    private final String message;

    private CouponStatus(Integer status, String message) {
        this.status = status;
        this.message = message;
    }
}

File: SmartWash-Android/app/src/main/java/com/smartwash/utils/CouponStatus.kt
package com.smartwash.utils

import androidx.annotation.StringRes
import com.smartwash.R

enum class CouponStatus(val status: String, @StringRes val descriptionRes: Int) {
    EXPIRED("1", R.string.coupon_status_expired),
    ACTIVE("0", R.string.coupon_status_active),
    RECEIVE("2", R.string.coupon_status_receive)
}
